controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -88,64 +94,73 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
         return Controller.NodeUri.newBuilder().setEndpoint(host.getIpAddr()).setPort(host.getPort()).build();
     }
 
+    private void closeConnection(Reply reply, RawClient client) {
+        log.info("Closing connection as a result of receiving: {}", reply);
+        if (client != null) {
+            try {
+                client.close();
+            } catch (Exception e) {
+                log.warn("Exception tearing down connection: ", e);
+            }
+        }
+    }
+
+    private boolean isReplyWanted(Reply reply, Class[] expectedReplies) {
+        for (Class<?> c : expectedReplies) {
+            if (c.isAssignableFrom(reply.getClass())) {
```

Comment:
instead of isAssignableFrom, we can do an exact match by sending a Set of expectedReplies and checking if set.contains(reply.class)

That will be more efficient and not require a loop.